### DISCUSSION TITLE
Erros object doesn't get updated with yup validation

### DISCUSSION THREAD

**Discussion by yishayweb** [react-hook-form/react-hook-form#8633] [Q&A] [answered]
I'm not sure if it's a bug, so I'm rasing it here first
I'm trying to validate a field based on the values of the validated field and another field
[CodeSandbox example](https://codesandbox.io/s/lively-resonance-lxndk3?file=/src/App.js)

In the example, I have **input1** and **input2**
I want to validate that **input2** length is shorter than **input1** and I want the validation error to be assigned to **input2**
When I'm doing changes to **input2** everything works as expected.
But when I change **input1** the errors object doesn't get updated even if it should

Flow to reproduce:
1. Insert "Hi" to input1
2. Insert "Hello" to input2 (The validation will work as expected now and the error message will appear)
3. Change input1 to be "Hi there"

After step 3 I would expect the errors objet to be updated and remove the error (because inpu1.length > input2.length), but it doesn't
Only after changing input2 again, the errors object will be updated

Is it possible to make the errors object to get updated even if I change only input1 ?

react-hook-form: 7.33.1
react: 18.0.0
@hookform/resolvers: 2.9.3
yup: 0.32.11

Thanks !!

**Comment by Moshyfawn** [comment, marked as answer]
Adding `{ deps: ['input2'] }` to the first input's `register` options seems to solve your use-case.

Check out [the `register` options RHF docs section](https://react-hook-form.com/api/useform/register#options) for more info on the `deps` API.

**Comment by yishayweb** [comment]
@moshyfawn thanks a lot for your quick response !!
Indeed it solves it
Do you know how can I do the same trick if I'm using **control** instead of **register** ?

**Reply by Moshyfawn** [reply]
You can use the same `deps` API with `Controller` / `useController` inside the `rules` prop:
```js
<Controller
  rules={{
    deps: ['input2']
  }}
/>
```

**Reply by yishayweb** [reply]
@Moshyfawn thank you so much !! You're a true hero :)

**Reply by Moshyfawn** [reply]
No problem, I'm glad I could help 💜
